tests/PHPStan/Parser/RichParserTest.php

```diff
@@ -267,47 +267,171 @@ public function dataLinesToIgnoreParseErrors(): iterable
 			'test(); // @phpstan-ignore ,' . PHP_EOL,
 			[
-				2 => ['First token is not an identifier'],
+				2 => ['Unexpected token type T_COMMA after start, expected T_IDENTIFIER'],
 			],
 		];
 
 		yield [
 			'<?php' . PHP_EOL .
 			'test(); // @phpstan-ignore return.ref, return.non )foo',
 			[
-				2 => ['Closing parenthesis ")" before opening parenthesis "("'],
+				2 => ['Unexpected token type T_CLOSE_PARENTHESIS after T_IDENTIFIER, expected T_COMMA or T_END or T_OPEN_PARENTHESIS'],
 			],
 		];
 
 		yield [
 			'<?php' . PHP_EOL .
 			'test(); // @phpstan-ignore return.ref, return.non (foo',
 			[
-				2 => ['Unclosed opening parenthesis "(" without closing parenthesis ")"'],
+				2 => ['Unexpected token type T_END, unclosed opening parenthesis'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore ()' . PHP_EOL,
+			[
+				2 => ['Unexpected token type T_OPEN_PARENTHESIS after start, expected T_IDENTIFIER'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore ' . PHP_EOL,
+			[
+				2 => ['Unexpected token type T_END after start, expected T_IDENTIFIER'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore ,' . PHP_EOL,
+			[
+				2 => ['Unexpected token type T_COMMA after start, expected T_IDENTIFIER'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore čumim' . PHP_EOL,
+			[
+				2 => ["Unexpected token type T_OTHER 'čumim' after start, expected T_IDENTIFIER"],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore test ((inner)' . PHP_EOL,
+			[
+				2 => ['Unexpected token type T_END, unclosed opening parenthesis'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore return.ref return.non',
+			[
+				2 => ['Unexpected token type T_IDENTIFIER after T_IDENTIFIER, expected T_COMMA or T_END or T_OPEN_PARENTHESIS'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore identifier (comment))' . PHP_EOL,
+			[
+				2 => ['Unexpected token type T_CLOSE_PARENTHESIS after T_CLOSE_PARENTHESIS, expected T_COMMA or T_END'],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore mečoun two' . PHP_EOL,
+			[
+				2 => ["Unexpected token type T_OTHER 'čoun' after T_IDENTIFIER, expected T_COMMA or T_END or T_OPEN_PARENTHESIS"],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore mečoun' . PHP_EOL,
+			[
+				2 => ["Unexpected token type T_OTHER 'čoun' after T_IDENTIFIER, expected T_COMMA or T_END or T_OPEN_PARENTHESIS"],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore return.ref, return.non čičí',
+			[
+				2 => ["Unexpected token type T_OTHER 'čičí' after T_IDENTIFIER, expected T_COMMA or T_END or T_OPEN_PARENTHESIS"],
+			],
+		];
+
+		yield [
+			'<?php' . PHP_EOL .
+			'test(); // @phpstan-ignore identifier -- comment' . PHP_EOL, // phpcs comment style
```

Comment:
Just a question - if the identifiers have to be delimited by comma, what are the reasons to not allow free-text comment, ie. comment not wrapped in parenthesis?

ex: `@phpstan-ignore return.reg https://github.com/doctrine/dbal/pull/6407`

Such grammar is unambiguous as after `@phpstan-ignore` and `,\s*` one identifier will be still required.

I find this more natural and it will also URL issue we currently have - `@phpstan-ignore return.reg (https://github.com/doctrine/dbal/pull/6407)` - is opened as `https://github.com/doctrine/dbal/pull/6407)`.

Reply:
With that, you could not use comma in comment, right?

`@phpstan-ignore identifier.one ignored, because of PHPStan bug, identifier.two`

Now you can (and you can even use parenthesis inside):

- `@phpstan-ignore identifier.one (ignored, because of PHPStan bug), identifier.two`
- `@phpstan-ignore identifier.one (ignored (because of PHPStan bug)), identifier.two`

Reply:
> @phpstan-ignore return.reg (https://github.com/doctrine/dbal/pull/6407) - is opened as https://github.com/doctrine/dbal/pull/6407).

Most widely used IDE (PHPStorm) does not do that (luckily for us).

Reply:
99% ignores are consisting of a single identifier. So I mean to parse anything after the current `\w+( \(...\))?(, \w+( \(...\))?)` (simplified) grammar as a free-text comment.